Source/Core/Core/Debugger/Thread.cpp

```diff
@@ -0,0 +1,159 @@
+// Copyright 2020 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#include "Core/Debugger/Thread.h"
+
+#include <algorithm>
+
+#include "Core/PowerPC/MMU.h"
+
+// Context offsets based on the following functions:
+//  - OSSaveContext
+//  - OSSaveFPUContext
+//  - OSDumpContext
+//  - OSClearContext
+//  - OSExceptionVector
+void Dolphin_Debugger::Thread::ReadContext(u32 addr, OSContext* ctx)
+{
+  for (int i = 0; i < 32; i += 1)
```

Comment:
the array sizes can be queried from the arrays themselves, eliminating the need for magic numbers here.

It's also our convention to use `i++` or `++i` when incrementing by 1 within an iteration expression